### PR TITLE
Python: Fix ql doc

### PR DESCRIPTION
**Pull request by yoff**
Fixes the non-`old` part of https://github.com/github/codeql-python-team/issues/191.

### CODE REVIEW

File: python/ql/src/semmle/python/dataflow/new/RemoteFlowSources.qll [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * Provides an extension point for for modelling user-controlled data.
```
Comment (tausbn) [inline python/ql/src/semmle/python/dataflow/new/RemoteFlowSources.qll:2]:
```suggestion
 * Provides an extension point for for modeling user-controlled data.
```
US English, alas...

File: python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll [hunk 2]
```diff
@@ -372,19 +377,27 @@ newtype TContent =
...
 class ListElementContent extends TListElementContent, Content {
   override string toString() { result = "List element" }
 }
 
+/** The content of an element of a set. */
 class SetElementContent extends TSetElementContent, Content {
   override string toString() { result = "Set element" }
 }
 
+/** The content, at a specifik index, of an element of a tuple. */
```
Comment (tausbn) [inline python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll:400]:
```suggestion
/** The content, at a specific index, of an element of a tuple. */
```

File: python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll [hunk 3]
```diff
@@ -352,7 +356,8 @@ class BarrierGuard extends GuardNode {
 }
 
 /**
- * A reference contained in an object. This is either a field or a property.
+ * IPA type for tracking data content associated with values.
```
Comment (RasmusWL) [inline python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll:359]: IPA type is an internal name. See https://help.semmle.com/QL/ql-handbook/types.html#algebraic-datatypes

```suggestion
 * Algebraic datatype for tracking data content associated with values.
```

File: python/ql/src/semmle/python/frameworks/MySQLdb.qll [hunk 4]
```diff
@@ -32,6 +32,7 @@ module MySQLdb {
   /** Gets a reference to the `MySQLdb` module. */
   DataFlow::Node moduleMySQLdb() { result = moduleMySQLdb(DataFlow::TypeTracker::end()) }
 
+  /** MySQLdb implements PEP 249, providing ways to execute SQL statments against a database. */
```
Comment (tausbn) [inline python/ql/src/semmle/python/frameworks/MySQLdb.qll:35]:
```suggestion
  /** MySQLdb implements PEP 249, providing ways to execute SQL statements against a database. */
```

**Review by tausbn — CHANGES_REQUESTED** [review]
A few typos, otherwise good stuff. :+1:

File: python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll [hunk 5]
```diff
@@ -372,19 +377,27 @@ newtype TContent =
...
+ * A data-flow value can have associated content.
+ * If the value is a collection, it can have elements,
+ * if it is an object, it can have attribute values.
+ */
 class Content extends TContent {
   /** Gets a textual representation of this element. */
   string toString() { result = "Content" }
 }
 
+/** The content of an element of a list. */
```
Comment (RasmusWL) [inline python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll:390]: I might be misunderstanding, but shouldn't this be `The element of a list`? (and likewise for the other QLDocs introduced in this PR).

If it's the content of an element of a list, for the code `[foo]` I would expect this to represent the _content of `foo`_, which I think is incorrect?

File: python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll (hunk 2 above)
Comment (RasmusWL) [inline python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll:400]:
```suggestion
/** The content, at a specific index, of an element of a tuple. */
```

**Review by RasmusWL — CHANGES_REQUESTED** [review]
Nice :+1: a few things I would like to have changed/clarified :blush:

File: python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll (hunk 5 above)
Comment (yoff) [reply python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll:390]: It depends on how you read "content of", I probably mean "value of", but I am happy to simply remove that possible ambiguity.

Comment (RasmusWL) [reply python/ql/src/semmle/python/dataflow/new/internal/DataFlowPublic.qll]: seems like we had this discussion before, so good that we could just reuse the old way of documenting this :smile:

**Review by RasmusWL — APPROVED** [review]
LGTM now :+1:

**Review by tausbn — APPROVED** [review]

**Pull request merged**